ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/NTupleOmitPrefixStorageLayoutExtension.java

```diff
@@ -0,0 +1,166 @@
+    
+    private static Pattern asciiPattern = Pattern.compile("\\A\\p{ASCII}*\\z");
+    private NTupleOmitPrefixStorageLayoutConfig config;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getExtensionName() {
+        return EXTENSION_NAME;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getDescription() {
+        return "This storage root extension describes an OCFL storage layout "
+        		+ "combining a pairtree-like root directory structure derived from "
+        		+ "prefix-omitted object identifiers, followed by the prefix-omitted "
+        		+ "object identifier themselves. The OCFL object identifiers are "
+        		+ "expected to contain prefixes which are removed in the mapping to "
+        		+ "directory names. The OCFL object identifier prefix is defined as "
+        		+ "all characters before and including a configurable delimiter. "
+        		+ "Where the prefix-omitted identifier length is less than "
+        		+ "tuple size * number of tuples, the remaining object id (prefix omitted) "
+        		+ "is left or right-side, zero-padded (configurable, left default), "
+        		+ "or not padded (none), and optionally reversed (default false). "
+        		+ "The object id is then divided into N n-tuple segments, and used "
+        		+ "to create nested paths under the OCFL storage root, followed by "
+        		+ "the prefix-omitted object id directory.";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void init(OcflExtensionConfig config) {
+    	Enforce.notNull(config, "configFile cannot be null");
+        
+    	if (!(config instanceof NTupleOmitPrefixStorageLayoutConfig)) {
+    		throw new OcflExtensionException(String.format("This extension only supports %s configuration. Received: %s",
+             	getExtensionConfigClass(), config));
+     	}
+
+    	NTupleOmitPrefixStorageLayoutConfig castConfig = (NTupleOmitPrefixStorageLayoutConfig) config;
+     	this.config = castConfig;
+    }
+
+
+    @Override
+    public Class<? extends OcflExtensionConfig> getExtensionConfigClass() {
+        return NTupleOmitPrefixStorageLayoutConfig.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String mapObjectId(String objectId) {
+        if (config == null) {
+            throw new OcflExtensionException("This extension must be initialized before it can be used.");
+        }
+        
+        if (!asciiPattern.matcher(objectId).matches()) {
+        	throw new OcflExtensionException(String.format("This id %s must contain only ASCII characters.", objectId));
+        }
+        //Split by delimiter and get the last part
+        String id = objectId.toLowerCase();
+        int index = id.lastIndexOf(config.getDelimiter().toLowerCase());
+        String section = objectId;
+        String baseObjectId = "";
```

Comment:
These do not need to be initialized because their value is either assigned on lines 123-124 or an exception is thrown.